flask/flask-launcher/src/main/java/net/corda/flask/launcher/Launcher.java

```diff
@@ -0,0 +1,170 @@
+package net.corda.flask.launcher;
+
+import lombok.SneakyThrows;
+import net.corda.flask.common.Flask;
+import net.corda.flask.common.LockFile;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.InputStream;
+import java.lang.reflect.Constructor;
+import java.net.URI;
+import java.net.URL;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.*;
+import java.util.jar.JarFile;
+import java.util.jar.Manifest;
+import java.util.stream.Collectors;
+import java.util.zip.ZipEntry;
+import java.util.zip.ZipFile;
+
+public class Launcher {
+
+    private static final String CACHE_FOLDER_DEFAULT_NAME = "flask_cache";
+    private static final Logger log = LoggerFactory.getLogger(Launcher.class);
+    private static final Path currentJar = findCurrentJar();
+
+    @SneakyThrows
+    private static Path findCurrentJar() {
+        String launcherClassName = Launcher.class.getName();
+        URL url = Launcher.class.getClassLoader().getResource(launcherClassName.replace('.', '/') + ".class");
+        if (url == null || !Objects.equals("jar", url.getProtocol()))
```

Comment:
You don't need `Objects.equals()` here:
```java
!"jar".equals(url.getProtocol())
```
is exactly the same because we _know_ `"jar"` is not `null`.